test/e2e/openshift/openshift_test.go

```diff
@@ -85,6 +86,38 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		}
 	})
 
+	It("should deploy all pods onto appropriate nodes", func() {
```

Comment:
Move your test at the end of the suite to ensure the pod list is not racing with the deployer pods and you are getting all pods we expect to run.

Reply:
Also I would reword the title of the test to "should not deploy infrastructure components in compute nodes" since you check just that (eg. you don't discriminate between pods that land in master vs infra).

Reply:
Ok, I will move this to the end of the tests.  Will also rename.